src/fidesops/ops/service/privacy_request/request_runner_service.py

```diff
@@ -291,13 +306,35 @@ async def run_privacy_request(
 
         except BaseException as exc:  # pylint: disable=broad-except
             privacy_request.error_processing(db=session)
-            # If dev mode, log traceback
+            # Send analytics to Fideslog
             await fideslog_graph_failure(
                 failed_graph_analytics_event(privacy_request, exc)
             )
+            # If dev mode, log traceback
             _log_exception(exc, config.dev_mode)
             return
 
+        # Send erasure requests via email to third parties where applicable
+        if can_run_checkpoint(
+            request_checkpoint=CurrentStep.erasure_email_post_send,
+            from_checkpoint=resume_step,
+        ):
+            try:
+                email_connector_erasure_send(
+                    db=session, privacy_request=privacy_request
+                )
+            except EmailDispatchException as exc:
+                privacy_request.cache_failed_checkpoint_details(
+                    step=CurrentStep.erasure_email_post_send, collection=None
+                )
+                privacy_request.error_processing(db=session)
+                await fideslog_graph_failure(
+                    failed_graph_analytics_event(privacy_request, exc)
+                )
+                # If dev mode, log traceback
+                _log_exception(exc, config.dev_mode)
+                return
+
```

Comment:
This is the primary change of this PR: if applicable, we combine cached data from all the collections on an email-based dataset and send a single email for each dataset.  There's no batching currently.  If there is no data to mask / the connection was read-only, etc. no data is sent.